query/src/sql/exec/data_schema_builder.rs

```diff
@@ -134,4 +135,44 @@ impl<'a> DataSchemaBuilder<'a> {
         }
         Ok(Arc::new(DataSchema::new(fields)))
     }
+
+    pub fn build_agg_func(
+        &self,
+        input_schema: DataSchemaRef,
+        exprs: &[Expression],
+    ) -> Result<(DataSchemaRef, Vec<Expression>)> {
+        let mut fields = input_schema.fields().clone();
+        let mut agg_inner_expressions = vec![];
+        for arg_expr in exprs.iter() {
+            match arg_expr {
+                Expression::AggregateFunction { args, .. } => {
+                    for arg_inner_expr in args.iter() {
```

Comment:
maybe need check args is not aggregate function? for example: `SUM(SUM(1))`;

Reply:
Nice!

Reply:
I thought this has been checked in planner? If not then you should also take care of the case `sum(func(sum(a)))`.

Reply:
It'll be checked in expression_chain, `Action must be a non-aggregated function`, but I think it's reasonable to check earlier

Reply:
> It'll be checked in expression_chain, `Action must be a non-aggregated function`, but I think it's reasonable to check earlier and give more meaningful error hint

You are right. Maybe it's better to fix this in another PR.